File: src/main/java/com/gusrylmubarok/reddit/backend/service/SubredditService.java
package com.gusrylmubarok.reddit.backend.service;

import com.gusrylmubarok.reddit.backend.dto.SubredditDto;
import java.util.List;

public interface SubredditService {
    SubredditDto save(SubredditDto subredditDto) throws Exception;
    List<SubredditDto> getAll();
    SubredditDto getSubreddit(Long id);
}

File: src/main/kotlin/com/reddot/service/SubredditService.kt
package com.reddot.service

import com.reddot.data.model.SubredditDTO

interface SubredditService {
    fun getAll(): List<SubredditDTO>
    fun getSubreddit(id: Long): SubredditDTO
    fun save(subredditDTO: SubredditDTO): SubredditDTO
}